vars/runATH.groovy

```diff
@@ -200,7 +200,25 @@ private void test(discriminator, commandBase, localSnapshots, localPluginsStashN
         realtimeJUnit(testResults: 'target/surefire-reports/TEST-*.xml', testDataPublishers: [[$class: 'AttachmentPublisher']]) {
             // Allow call old images without the set-java.sh script. A message is showed if it doesn't exist to alert
             // that you are using a new pipeline-library with an old ATH source/image (... type: ./set-java.sh: not found)
-            def command = 'if type ./set-java.sh; then ./set-java.sh $java_version; fi; eval "$(./vnc.sh)" && export DISPLAY=$BROWSER_DISPLAY && export SHARED_DOCKER_SERVICE=true && export EXERCISEDPLUGINREPORTER=textfile && ' + prepareCommand(commandBase, discriminator, localSnapshots, localPluginsStashName)
+            def command = '''
+            if [ -x ./set-java.sh ]; then
+                ./set-java.sh $java_version;
+            else
+                if [ $java_version -gt 8 ]; then
+                    echo "Info: ./set-java.sh not found because you are using old ATH sources, $java_version cannot be used, instead java 8 will be used"
```

Comment:
Can we fail the step here instead so problems gets the attention they need?

Reply:
I agree with Oliver here, if we are using incoherent versions hidding the thing in a line of log is going to make much harder to find the issue when everything blow up in core. Failing by design in this case is going to make the problem evident and force users to fix it

Reply:
We can also mark the build as UNSTABLE if set-java.sh is not found even when running on java8.

Reply:
Well, we know what java version did the user wish to use and we know whether the switcher exists which leaves us with 4 possibilities:

switcher exists / java version | 8 | 11
-----------------------------------------|---|----
yes | switch to it | switch to it
no | java 8 will be used | _unspecified java used_

I believe it has to fail in case of java 11 and no switcher present. The question is if we want to rely the absence of the switcher file in the container is a sufficient guarantee the java version is 8.

What @batmat is suggesting would probably work, though I do not like the way it complicates the pipeline-library for such a minor use-case.

Reply:
No we have:

|set-java.sh exists / java_version specified | 8 (or null) | 11
|---|---|---|
| Yes |	SUCCESS | SUCCESS |
| No | SUCCESS (1) | FAILURE |

My proposal is:
(1) What about changing SUCCESS to UNSTABLE in this case

If we keep it as SUCCESS, the user is not aware of a new version of ATH, and (s)he's not going to update it. We only write this in the log:
`INFO: ./set-java.sh not found because you are using old ATH sources, please consider to update ATH sources and docker image`
And most likely the admin is not going to review all the log messages, as the build was SUCCESS.

But doing the result as UNSTABLE, (s)he become more conscious. 

WDYT? _SUCCESS_ or _UNSTABLE_?

Reply:
Can we use **githubNotify** to add this info in the GH status, instead of using the job status? 

Actually a @daniel-beck's proposal :+1:

Reply:
I do not feel strongly. What you say makes sense.

Reply:
Decision at: https://github.com/jenkins-infra/pipeline-library/pull/84#pullrequestreview-189421087